src/sources/kafka.rs

```diff
@@ -392,28 +518,36 @@ mod integration_test {
             config.headers_key,
             ShutdownSignal::noop(),
             tx,
+            acknowledgements,
         ));
-        let events = collect_n(rx, 1).await;
-        println!("Received event  {:?}", events[0].as_log());
-
-        assert_eq!(
-            events[0].as_log()[log_schema().message_key()],
-            "my message".into()
-        );
-        assert_eq!(events[0].as_log()["message_key"], "my key".into());
-        assert_eq!(
-            events[0].as_log()[log_schema().source_type_key()],
-            "kafka".into()
-        );
-        assert_eq!(
-            events[0].as_log()[log_schema().timestamp_key()],
-            now.trunc_subsecs(3).into()
-        );
-        assert_eq!(events[0].as_log()["topic"], topic.into());
-        assert!(events[0].as_log().contains("partition"));
-        assert!(events[0].as_log().contains("offset"));
-        let mut expected_headers = BTreeMap::new();
-        expected_headers.insert("my header".to_string(), Value::from("my header value"));
-        assert_eq!(events[0].as_log()["headers"], Value::from(expected_headers));
+        let events = collect_n(rx, 10).await;
+
+        let client: BaseConsumer = client_config();
+        let watermarks = client
+            .fetch_watermarks(&topic, 0, Duration::from_secs(1))
```

Comment:
This sometimes dies with an "invalid partition" error. However, shouldn't there only be one partition (numbered `0`)? Any ideas what's going on there?